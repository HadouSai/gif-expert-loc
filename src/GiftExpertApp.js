import React, { useState } from "react";
import PropTypes from "prop-types";
import AddCategory from "./components/AddCategory";
import GiffGrid from "./components/GiffGrid";

/**
 *
 * @param {*} initialCategories `[]`
 * @returns
 */
const GiftExpertApp = ({ initialCategories }) => {
  //const category = ['Morgana', 'Gwen', 'Seraphine'];
  const [categories, setCategories] = useState(initialCategories);
  //const newCategory = 'randomNyx';

  return (
    <div>
      <h2>GiftExpertApp</h2>

      <AddCategory setCategories={setCategories} />
      <hr />

      <div>
        {categories.map((category, i) => (
          <GiffGrid key={i} category={category} />
        ))}
      </div>
    </div>
  );
};

GiftExpertApp.propTypes = {
  initialCategories: PropTypes.array,
};
GiftExpertApp.defaultProps = {
  initialCategories: ["Hatsune"],
};

export default GiftExpertApp;
