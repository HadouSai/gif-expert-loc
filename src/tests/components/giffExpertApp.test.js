import "@testing-library/jest-dom";
import { shallow } from "enzyme"
import GiftExpertApp from "../../GiftExpertApp";

describe('testing in main component', () => {
    test('should appear in root', () => {
        const wrapper = shallow(<GiftExpertApp />);
        expect(wrapper).toMatchSnapshot();
    })

})
