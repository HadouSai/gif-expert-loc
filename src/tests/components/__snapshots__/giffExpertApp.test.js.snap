// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing in main component should appear in root 1`] = `
<div>
  <h2>
    GiftExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <div>
    <GiffGrid
      category="Hatsune"
      key="0"
    />
  </div>
</div>
`;
